api/resolvers/item.js

```diff
@@ -812,6 +812,16 @@ export default {
             item: updatedItem,
             tag: `TIP-${updatedItem.id}`
           })
+
+          // send push notifications to forwarded recipients
+          if (mappedForwards.length) {
+            await mappedForwards.map(forward => sendUserNotification(forward.user.id, {
```

Comment:
is this not missing a `Promise.allSettled`?

Reply:
Eek yep I believe it is. I'll correct that tomorrow

Reply:
> since we're not waiting on the successful notification sends for anything further in this function

to be pedantic, I believe the `catch` will only work with a proper `await` (so with `Promise.allSettled`). However, not 100% sure on that :)

Reply:
I think if you use `await` on a non-promise, it just returns an immediately-resolved promise, resolved with the `await`ed value, so yea it wouldn't catch exceptions from the inner promises.

Also, `Promise.allSettled` won't reject, even if any of the contained promises reject, so `await`ing `Promise.allSettled` also won't trigger the `catch` block, which is what I originally meant. We'd have to iterate through the results and see if any of the statuses were `rejected` and log them accordingly.

Interesting discussion 🍻

Reply:
> Also, Promise.allSettled won't reject, even if any of the contained promises reject, so awaiting Promise.allSettled also won't trigger the catch block, which is what I originally meant. We'd have to iterate through the results and see if any of the statuses were rejected and log them accordingly.

Haha, I will just pretend that this is what I meant with

> However, not 100% sure on that in this case :)

lol

> Interesting discussion 🍻

Yes :) I love nerding out, lol